checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsTransfer.java

```diff
@@ -33,6 +58,72 @@ public CalledMethodsTransfer(final CFAnalysis analysis) {
+        new ConditionalTransferResult<>(
+            result.getResultValue(),
+            result.getThenStore(),
+            result.getElseStore(),
+            exceptionalStores);
+    exceptionalStores = null;
+    return finalResult;
+  }
+
+  @Override
+  public void accumulate(Node node, TransferResult<CFValue, CFStore> result, String... values) {
+    super.accumulate(node, result, values);
+    if (exceptionalStores == null) {
+      return;
+    }
+
+    List<String> valuesAsList = Arrays.asList(values);
+    // If dataflow has already recorded information about the target, fetch it and integrate
+    // it into the list of values in the new annotation.
+    JavaExpression target = JavaExpression.fromNode(node);
+    if (CFAbstractStore.canInsertJavaExpression(target)) {
+      CFValue flowValue = result.getRegularStore().getValue(target);
+      if (flowValue != null) {
+        Set<AnnotationMirror> flowAnnos = flowValue.getAnnotations();
+        assert flowAnnos.size() <= 1;
+        for (AnnotationMirror anno : flowAnnos) {
+          if (atypeFactory.isAccumulatorAnnotation(anno)) {
+            List<String> oldFlowValues =
+                AnnotationUtils.getElementValueArray(
+                    anno,
+                    ((CalledMethodsAnnotatedTypeFactory) atypeFactory).calledMethodsValueElement,
+                    String.class);
+            // valuesAsList cannot have its length changed -- it is backed by an
+            // array.  getElementValueArray returns a new, modifiable list.
+            oldFlowValues.addAll(valuesAsList);
+            valuesAsList = oldFlowValues;
+          }
+        }
+      }
+      AnnotationMirror newAnno = atypeFactory.createAccumulatorAnnotation(valuesAsList);
+      exceptionalStores.values().stream().forEach(s -> s.insertValue(target, newAnno));
+    }
+  }
+
+  /**
+   * Create a set of stores for the exceptional paths out of the block containing {@code node}, to
+   * allow the fact that the method being invoked in {@code node} was definitely called to be
+   * propagated along those paths.
+   *
+   * @param node a method invocation
+   * @param input the transfer input associated with the method invocation
+   * @return a map from types to stores. The keys are the same keys used by {@link
+   *     ExceptionBlock#getExceptionalSuccessors()}. The values are copies of the regular store from
+   *     {@code input}.
+   */
+  private Map<TypeMirror, CFStore> makeExceptionalStores(
+      MethodInvocationNode node, final TransferInput<CFValue, CFStore> input) {
+    if (!(node.getBlock() instanceof ExceptionBlock)) {
+      // this can happen in some weird (buggy?) cases
+      return Collections.emptyMap();
+    }
+    ExceptionBlock block = (ExceptionBlock) node.getBlock();
+    Map<TypeMirror, CFStore> result = new LinkedHashMap<>();
+    block.getExceptionalSuccessors().keySet().stream()
+        .forEach(tm -> result.put(tm, input.getRegularStore().copy()));
```

Comment:
Rather than using Java streams, it is simpler and more efficient to use `Map.forEach()`.  Please do that throughout.